src/App.js

```diff
@@ -8,28 +8,39 @@ const useSemiPersistentState = (state) =>{
   const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem(state))||[]);
   useEffect(()=>{
     localStorage.setItem(state, JSON.stringify(todoList))
-}, [todoList])
+}, [todoList, state])
 return [todoList,setTodoList]
 }
 
 function App() { 
   const [todoList, setTodoList] = useSemiPersistentState("savedTodoList");
-  console.log('TODO: ', todoList);
+
   const addTodo = (newTodo) =>{
-    // setTodoList([...todoList ,newTodo])
     setTodoList((todoList) => [...todoList, newTodo])
   }
+
+  function removeTodo(id) {
+    console.log('index', id);
+    const index = todoList.findIndex(todo => todo === id);
+    if (index !== -1) {
+      const newTodoList = [...todoList];
+      newTodoList.splice(index, 1);
+      setTodoList(newTodoList);
+    }
```

Comment:
Awesome job here!! 💯 
But Remember how we got the previous todoList state in `addTodo`. You should do the same in `removeTodo`.
Also check out the `filter()` function on a list
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
👍